backend/data/users.js

```diff
@@ -301,6 +301,27 @@ async function addGrade(studentId, courseId, assignmentId, grade) {
   return { gradeAdded: true };
 }
 
+/**
+ * Fetch grade of a student by assignmentId
+ * @param {string} assignmentId 
+ * @param {string} studentId 
+ * @returns object
+ */
+async function fetchGrade(assignmentId, studentId) {
+  const errorMSg = 'No assignments found for the given assignment id';
+  const user = await getUser(studentId);
+  if (user && !user.classes) throw new Error(errorMSg);
+  for (const classObj of user.classes) {
+    const assignment = classObj.grades.find((e) => e._id === assignmentId);
```

Comment:
```suggestion
    const assignment = classObj.grades.find((e) => e._id.toString() === assignmentId);
```

We would need to toString() the id from the database since we store all ids as ObjectID

Reply:
According to the current schema, id's are stored as strings.